packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2470,27 +2470,77 @@ class RenderRepaintBoundary extends RenderProxyBox {
   /// will give you a 1:1 mapping between logical pixels and the output pixels
   /// in the image.
   ///
+  /// ## Sample code
+  ///
+  /// The following is an example of how to go from a `GlobalKey` on a
+  /// `RepaintBoundary` to a PNG:
+  ///
+  /// ```dart
+  /// class PngApp extends StatelessWidget {
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return new MaterialApp(
+  ///       title: 'Flutter Demo',
+  ///       theme: new ThemeData(
+  ///         primarySwatch: Colors.blue,
+  ///       ),
+  ///       home: new PngHome(title: 'PNG Demo'),
+  ///     );
+  ///   }
+  /// }
+  ///
+  /// class PngHome extends StatefulWidget {
+  ///   PngHome({Key key, this.title}) : super(key: key);
+  ///
+  ///   final String title;
+  ///
+  ///   @override
+  ///   _PngHomeState createState() => new _PngHomeState();
+  /// }
+  ///
+  /// class _PngHomeState extends State<PngHome> {
+  ///   GlobalKey globalKey = new GlobalKey();
+  ///
+  ///   Future<void> _capturePng() async {
+  ///     RenderRepaintBoundary boundary = globalKey.currentContext.findRenderObject();
+  ///     ui.Image image = await boundary.toImage();
+  ///     ByteData byteData = await image.toByteData(format: ui.ImageByteFormat.png);
+  ///     Uint8List pngBytes = byteData.buffer.asUint8List();
+  ///     print(pngBytes);
+  ///   }
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return Scaffold(
+  ///       appBar: new AppBar(
+  ///         title: new Text(widget.title),
+  ///       ),
+  ///       body: RepaintBoundary(
+  ///         key: globalKey,
+  ///         child: const Center(
+  ///           child: Text(
+  ///             'Hello World',
+  ///             textDirection: TextDirection.ltr,
+  ///           ),
+  ///         ),
+  ///       ),
+  ///       floatingActionButton: FloatingActionButton(
+  ///         child: Icon(Icons.picture_in_picture),
+  ///         tooltip: 'Capture PNG',
+  ///         onPressed: _capturePng,
+  ///       ),
+  ///     );
```

Comment:
to keep this briefer, remove the scaffold, leaving only the repaint boundary, and replace the text widget with a FlatButton that does what the FAB is doing in your example. So it's a button that takes a screenshot of itself.